core/src/main/java/org/jboss/pnc/build/finder/core/LocalFile.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2017 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.pnc.build.finder.core;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Set;
+
+import org.infinispan.commons.marshall.AdvancedExternalizer;
+import org.infinispan.commons.marshall.SerializeWith;
+import org.infinispan.commons.util.Util;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+
+@JsonDeserialize
+@SerializeWith(LocalFile.LocalFileExternalizer.class)
+public class LocalFile {
+    private final String filename;
+
+    private final long size;
+
+    @JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
+    public LocalFile(@JsonProperty("filename") String filename, @JsonProperty("size") long size) {
+        this.filename = filename;
+        this.size = size;
+    }
+
+    public String getFilename() {
+        return filename;
+    }
+
+    public long getSize() {
+        return size;
+    }
+
+    @Override
+    public String toString() {
+        return "LocalFile{" + "filename='" + filename + '\'' + ", size=" + size + '}';
+    }
+
+    public static class LocalFileExternalizer implements AdvancedExternalizer<LocalFile> {
+        private static final long serialVersionUID = -3322870654564600039L;
+
+        private static final Integer ID = (Character.getNumericValue('L') << 16) | (Character.getNumericValue('F') << 8)
+                | Character.getNumericValue('E');
+
+        private static final int VERSION = 1;
+
+        @Override
+        public void writeObject(ObjectOutput output, LocalFile object) throws IOException {
+            output.writeInt(VERSION);
```

Comment:
If you look at `kojiji/src/main/java/com/redhat/red/build/koji/model/xmlrpc/KojiBuildInfo.java`, this is done a little differently with `implements Externalizable` and not `implements AdvancedExternalizer<>`. What is the difference?

Reply:
If you want to have your classes serialized in ISPN Cache, you create a protobuf schema (can be generated or manually created) or use JbossMarshaller(which build-finder uses) and register AdvancedExternalizers(for every class that is directly stored in cache, for subclasses like KojiBuildInfo Externalizable is enough). 

For example you've implemented AdvancedExternalizer in KojiBuild [1] which is directly stored in the cache and called writeObject on each subclass like KojiBuildInfo, KojiTaskInfo etc... The writeObject then uses implementation of Externalizable of these subclasses.

The registration of AE was done here [2]


[1] https://github.com/project-ncl/build-finder/blob/6f58217fd76440c892ca13a197a6d2f1b38d0281/core/src/main/java/org/jboss/pnc/build/finder/koji/KojiBuild.java#L307
[2] https://github.com/project-ncl/build-finder/blob/6f58217fd76440c892ca13a197a6d2f1b38d0281/core/src/main/java/org/jboss/pnc/build/finder/koji/KojiBuild.java#L307

Reply:
I am mixing up two things: before I could even use the `AdvancedExternalizer`, I had to submit a PR to the kojiji code to make those objects `Serializable` first. Perhaps this is the code I was thinking of.

The code in `KojiBuild.java` was only a few lines, so when I saw your code I think that I was surprised by the number of lines required.

Note that this method of serialization (not using protobuf) has been deprecated now for a while, but this code is easier to write, I think, and I didn't want to have to rewrite everything, including adding new code for protobuf into kojiji itself.

Reply:
@dwalluck Happens :). 

Anyway, you are right. Using JBossMarshaller is deprecated. I've got experience working with protobuf in ISSPN and I could look into implementing it into build-finder.